src/celeritas/optical/model/RayleighModel.cc

```diff
@@ -0,0 +1,100 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/model/RayleighModel.cc
+//---------------------------------------------------------------------------//
+#include "RayleighModel.hh"
+
+#include "corecel/Assert.hh"
+#include "corecel/io/Logger.hh"
+#include "celeritas/io/ImportOpticalMaterial.hh"
+#include "celeritas/optical/MaterialParams.hh"
+#include "celeritas/optical/MfpBuilder.hh"
+
+#include "RayleighMfpCalculator.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct the model from imported data.
+ */
+RayleighModel::RayleighModel(ActionId id,
+                             SPConstImported imported,
+                             SPConstMaterials materials)
+    : Model(id, "optical-rayleigh", "interact by optical Rayleigh")
+    , imported_(ImportModelClass::rayleigh, imported)
+    , materials_(std::move(materials))
+{
+    CELER_EXPECT(materials_);
+    CELER_EXPECT(materials_->num_materials() == imported_.num_materials());
+
+    for (auto mat : range(OpticalMaterialId(materials_->num_materials())))
+    {
+        CELER_EXPECT(imported_.mfp(mat) || materials_->rayleigh_material(mat));
+    }
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Build the mean free paths for the model.
+ */
+void RayleighModel::build_mfps(OpticalMaterialId mat, MfpBuilder& build) const
+{
+    CELER_EXPECT(mat < materials_->num_materials());
+
+    if (auto const& mfp = imported_.mfp(mat))
+    {
+        build(mfp);
+    }
+    else
+    {
+        RayleighMfpCalculator calc_mfp(
+            MaterialView(materials_->host_ref(), mat),
+            materials_->rayleigh_material(mat));
+
+        // Use index of refraction energy grid as calculated MFP energy grid
+        auto const& energy_grid = calc_mfp.grid();
+
+        ImportPhysicsVector result{ImportPhysicsVectorType::free,
+                                   std::vector<double>(energy_grid.size()),
+                                   std::vector<double>(energy_grid.size())};
+
+        for (auto i : range(energy_grid.size()))
+        {
+            result.x[i] = energy_grid[i];
+            result.y[i]
+                = calc_mfp(native_value_to<units::MevEnergy>(result.x[i]));
+        }
+
+        build(result);
```

Comment:
Perhaps instead of creating an `ImportPhysicsVector` here we could use the `GenericGridInserter::operator()` that takes two spans, and just create a vector of the calculated mfps? And if we added a method to get a span of all the grid values from the `NonuniformGrid`, something like:
```c++
template<class T>
CELER_FUNCTION auto NonuniformGrid<T>::values() const -> SpanT
{
    return storage_[offset_];
}
```
then we wouldn't have to copy the grid values either.